contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -1024,153 +1000,53 @@ contract LoanTokenLogicStandard is AdvancedToken {
         return IERC20(loanTokenAddress).balanceOf(address(this));
     }
 
-    /* Internal View functions */
-
-    function _tokenPrice(
-        uint256 assetSupply)
-        internal
-        view
-        returns (uint256)
-    {
-        uint256 totalTokenSupply = totalSupply_;
-
-        return totalTokenSupply != 0 ?
-            assetSupply
-                .mul(WEI_PRECISION)
-                .div(totalTokenSupply) : initialPrice;
-    }
-
-    function _avgBorrowInterestRate(
-        uint256 assetBorrow)
-        internal
-        view
-        returns (uint256)
-    {
-        if (assetBorrow != 0) {
-            (uint256 interestOwedPerDay,) = _getAllInterest();
-            return interestOwedPerDay
-                .mul(365 * WEI_PERCENT_PRECISION)
-                .div(assetBorrow);
-        }
-    }
-
-    // next supply interest adjustment
-    function _supplyInterestRate(
+    function _nextSupplyInterestRate(
+        uint256 nextBorrowRate,
         uint256 assetBorrow,
         uint256 assetSupply)
-        internal
+        public
         view
         returns (uint256)
     {
         if (assetBorrow != 0 && assetSupply >= assetBorrow) {
-            return _avgBorrowInterestRate(assetBorrow)
+            return nextBorrowRate
                 .mul(_utilizationRate(assetBorrow, assetSupply))
                 .mul(SafeMath.sub(WEI_PERCENT_PRECISION, IBZx(bZxContract).lendingFeePercent()))
                 .div(WEI_PERCENT_PRECISION * WEI_PERCENT_PRECISION);
         }
     }
 
     function _nextBorrowInterestRate(
```

Comment:
I recommend renaming this to nextBorrowInterestRate and on the interface add it as an overloaded function as the _ prefix should be reserved for internal functions only. I can understand why it makes sense to have both this and the auto filling version as publicly accessible.

Reply:
this is internal but public to allow protocol to call it.